ffi/capi/dart/package/lib/lib.dart

```diff
@@ -0,0 +1,11466 @@
+class _CodePointRangeIteratorResultFfi extends ffi.Struct {
+  @ffi.Uint32()
+  external int start;
+  @ffi.Uint32()
+  external int end;
+  @ffi.Bool()
+  external bool done;
+}
+
+class CodePointRangeIteratorResult {
+  final _CodePointRangeIteratorResultFfi _underlying;
+
+  // ignore: unused_element
+  CodePointRangeIteratorResult._(this._underlying);
+
+  factory CodePointRangeIteratorResult() {
+    final pointer = ffi2.calloc<_CodePointRangeIteratorResultFfi>();
+    final result = CodePointRangeIteratorResult._(pointer.ref);
+    _callocFree.attach(result, pointer.cast());
+    return result;
+  }
+
+  int get start => _underlying.start;
+  set start(int start) {
+    _underlying.start = start;
+  }
+
+  int get end => _underlying.end;
+  set end(int end) {
+    _underlying.end = end;
+  }
+
+  bool get done => _underlying.done;
+  set done(bool done) {
+    _underlying.done = done;
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is CodePointRangeIteratorResult &&
+      other._underlying.start == _underlying.start &&
+      other._underlying.end == _underlying.end &&
+      other._underlying.done == _underlying.done;
+
+  @override
+  int get hashCode => Object.hashAll([
+        _underlying.start,
+        _underlying.end,
+        _underlying.done,
+      ]);
+}
+
+/// The various calendar types currently supported by [`ICU4XCalendar`]
+///
+/// See the [Rust documentation for `AnyCalendarKind`](https://docs.rs/icu/latest/icu/calendar/enum.AnyCalendarKind.html) for more information.
+enum ICU4XAnyCalendarKind {
+  /// The kind of an Iso calendar
+  iso.__(0),
+
+  /// The kind of a Gregorian calendar
+  gregorian.__(1),
+
+  /// The kind of a Buddhist calendar
+  buddhist.__(2),
+
+  /// The kind of a Japanese calendar with modern eras
+  japanese.__(3),
+
+  /// The kind of a Japanese calendar with modern and historic eras
+  japaneseExtended.__(4),
+
+  /// The kind of an Ethiopian calendar, with Amete Mihret era
+  ethiopian.__(5),
+
+  /// The kind of an Ethiopian calendar, with Amete Alem era
+  ethiopianAmeteAlem.__(6),
+
+  /// The kind of a Indian calendar
+  indian.__(7),
+
+  /// The kind of a Coptic calendar
+  coptic.__(8),
+
+  /// The kind of a Dangi calendar
+  dangi.__(9),
+
+  /// The kind of a Chinese calendar
+  chinese.__(10),
+
+  /// The kind of a Hebrew calendar
+  hebrew.__(11),
+
+  /// The kind of a Islamic civil calendar
+  islamicCivil.__(12),
+
+  /// The kind of a Islamic observational calendar
+  islamicObservational.__(13),
+
+  /// The kind of a Islamic tabular calendar
+  islamicTabular.__(14),
+
+  /// The kind of a Islamic Umm al-Qura calendar
+  islamicUmmAlQura.__(15),
+
+  /// The kind of a Persian calendar
+  persian.__(16),
+
+  /// The kind of a Roc calendar
+  roc.__(17);
+
+  const ICU4XAnyCalendarKind.__(this._id);
+
+  // ignore: unused_element
+  factory ICU4XAnyCalendarKind._(int id) =>
+      values.firstWhere((value) => value._id == id);
+
+  final int _id;
```

Comment:
Is the `_id` for `enum`s always just the `index`? If so, it could be omitted.

Reply:
`index`?

Reply:
For types liks `ICU4XError` the ids are non continuous...

Reply:
In general a Diplomat enum's values are arbitrary. I don't think complicating the generation code to special case the continuous case is worth the effort.

Reply:
Done